backtrace-sys/build.rs

```diff
@@ -89,23 +122,17 @@ fn main() {
 
     let cfg = cc::Build::new();
     let compiler = cfg.get_compiler();
-    let cc = compiler.path().file_name().unwrap().to_str().unwrap();
-    let mut flags = OsString::new();
-    for (i, flag) in compiler.args().iter().enumerate() {
-        if i > 0 {
-            flags.push(" ");
-        }
-        flags.push(flag);
-    }
-    let ar = find_tool(&compiler, cc, "ar");
-    let ranlib = find_tool(&compiler, cc, "ranlib");
+    let cc = compiler_cc_variable(&compiler);
+    let flags = compiler_flags(&compiler);
+    let ar = find_tool(&compiler, &cc, "ar");
+    let ranlib = find_tool(&compiler, &cc, "ranlib");
     let mut cmd = Command::new("sh");
 
     cmd.arg(configure)
        .current_dir(&dst)
        .env("AR", &ar)
        .env("RANLIB", &ranlib)
-       .env("CC", compiler.path())
+       .env("CC", cc.clone())
```

Comment:
I wonder if we can just fold all the flags into this env var? Something like setting `CC="sccache gcc"` for the configure script as well?

Reply:
You mean something like 

```rust
let mut cc_with_flags = compiler.path().to_path_buf().into_os_string();
cc_with_flags.push(" ");
cc_with_flags.push(flags);
```
And then
```rust
    .env("CC", &cc_with_flags)
    .env("CFLAGS", "")
```
?

It works and its definitely less code! But not sure about different platforms or something. 

Just as idea - maybe make function to get CC and CFLAGS from cc-rs's ```Tool``` and add it to ```cc-rs```? It may be usefull for such cases.

Reply:
Yeah that's what I was thinking, although this may or may not be one of those build systems that rejects such a pattern.

Although along the lines of modifying the `cc` crate I wonder if we could perhaps fix this upstream? The `cc` crate knows when it recognizes ccache/sccache and whatnot, so I wonder if we could make `Tool::path` return the actual compiler, not the ccache path? That way all tools looking at paths and such would "just work", but compilations using `cc` would also use sccache?

Reply:
As for me - it would be a bit confusing for a user - some crates will use ccache and some will not.

I think the root of the problem - that some of build systems ignore compiler arguments sometimes, so we cant use compiler path as a compilation command. 

Maybe something like this will work? 

when compiling with CC="ccache gcc"  ```cc``` would return "gcc" as compiler path and flags vec without "gcc" in the first element. But at the same time, it will have ```.cc()``` and ```.cflags()``` methods just for that case. And those methods may be used for other env-variables based build systems.

Reply:
Yeah that strategy sound plausible to me!